src/txn/txn_rollback_to_stable.c

```diff
@@ -960,6 +960,88 @@ __rollback_to_stable_btree(WT_SESSION_IMPL *session, wt_timestamp_t rollback_tim
     return (ret);
 }
 
+/*
+ * __rollback_evict_exclusive_apply --
+ *     Apply an eviction helper to all dhandles.
+ */
+static int
+__rollback_evict_exclusive_apply(WT_SESSION_IMPL *session, int (*dhandle_func)(WT_SESSION_IMPL *))
+{
+    WT_CURSOR *metadata_cursor;
+    WT_DECL_RET;
+    const char *uri;
+
+    WT_RET(__wt_metadata_cursor(session, &metadata_cursor));
```

Comment:
I think this loop should be over the dhandles similarly to how the checkpoint walks the list of dhandles (or probably more specifically similarly to how the sweep thread walks dhandles, since we don't care about seeing a consistent view of all handles for this check). We don't need to consider handles that aren't currently open - we can't possibly be doing an eviction on those handles.

Reply:
If a dhandle gets opened and an eviction happens, it will make a transaction and will cause the transaction check to fail.

Reply:
That shouldn't happen, because applications aren't doing anything while also doing a rollback-to-stable. This is really just a check to help applications execute on that requirement - we don't expect to be able to do a perfect job because we don't have a mechanism for blocking other operations.

Reply:
Why though? If an application is running rollback to stable, eviction threads may still be active.

This code is stopping eviction so it doesn't interfere with the check. I agree the check doesn't have to be perfect but the stopping of eviction **must** be perfect otherwise we will get transient failures in rollback to stable.

Reply:
The difference here is significant. It is common enough for a user to have 100's of thousands of handles, but very uncommon for them to have all handles open. Forcing rollback to stable to iterate over and open all dhandles isn't useful work, since a handle that isn't open can't possibly be having any content evicted from it.

I also suspect this approach will encounter EBUSY returns in some situations that will be difficult to avoid generally.